### PR TITLE
Fine tunning Types module [12963]

### PR DESCRIPTION
**Pull request by jparisu**
Adding comments, logs, delete functions, etc.
Classes: Endpoint, Guid, QoS, Topic, Data, ParticipantId, ParticipantType, TimeStamp, utils

Merge after #49

### CODE REVIEW

File: test/unittest/configuration/ConfigurationTest.cpp [hunk 1]
```diff
@@ -213,6 +213,19 @@ std::string random_participant_name(
     return std::string("PartName_") + std::to_string(seed);
 }
 
+/*
+ * Random participant type
+ *
+ * WARNING: the max_types_available must be updated with each new type added
```
Comment (juanlofer-eprosima) [inline test/unittest/configuration/ConfigurationTest.cpp:219]: Maybe in the future make ParticipantType::participant_type_with_name_ public so it is accessible from here.

**Review by juanlofer-eprosima — CHANGES_REQUESTED** [review]
Please fix uncrustify

File: include/ddsrouter/types/RawConfiguration.hpp [hunk 2]
```diff
@@ -39,6 +39,8 @@ using RawConfigurationType = YAML::NodeType;
  *
  * @param [in] file_path : path of file to load
  * @return yaml in RawConfiguration format contained in file_path
+ *
+ * @throw \c ConfigurationException in case the file does not exist or it is not a correct yaml
```
[the remark this answers is not in the document]
Comment (jparisu) [reply include/ddsrouter/types/RawConfiguration.hpp:43]: I think correct is more accurate. The yaml is not needed to be "valid yaml", is needed to have a specific format, so I think it must be "correct".

File: include/ddsrouter/types/configuration_tags.hpp [hunk 3]
```diff
@@ -28,12 +28,12 @@ namespace ddsrouter {
...
 
 // Topics related tags
-constexpr const char* ALLOWLIST_TAG("allowlist");
-constexpr const char* BLOCKLIST_TAG("blocklist");
-constexpr const char* TOPIC_NAME_TAG("name");
-constexpr const char* TOPIC_TYPE_NAME_TAG("type");
+constexpr const char* ALLOWLIST_TAG("allowlist");   //! List of allowed topics
+constexpr const char* BLOCKLIST_TAG("blocklist");   //! List of blocked topics
+constexpr const char* TOPIC_NAME_TAG("name");       //! Name of a topic
+constexpr const char* TOPIC_TYPE_NAME_TAG("type");  //! Name of type of a topic
```
[the remark this answers is not in the document]
Comment (jparisu) [reply include/ddsrouter/types/configuration_tags.hpp:34]: :+1:

File: include/ddsrouter/types/configuration_tags.hpp [hunk 4]
```diff
@@ -28,12 +28,12 @@ namespace ddsrouter {
...
 
-constexpr const char* PARTICIPANT_TYPE_TAG("type");
+constexpr const char* PARTICIPANT_TYPE_TAG("type"); //! Type of participant
```
[the remark this answers is not in the document]
Comment (jparisu) [reply include/ddsrouter/types/configuration_tags.hpp:36]: :+1:

File: include/ddsrouter/writer/implementations/auxiliar/DummyWriter.hpp [hunk 5]
```diff
@@ -21,9 +21,11 @@
 
 #include <mutex>
 
-#include <ddsrouter/types/ParticipantId.hpp>
+#include <ddsrouter/types/participant/ParticipantId.hpp>
 #include <ddsrouter/types/Timestamp.hpp>
 #include <ddsrouter/writer/implementations/auxiliar/EchoWriter.hpp>
+#include <ddsrouter/writer/implementations/auxiliar/EchoWriter.hpp>
+#include <ddsrouter/writer/IWriter.hpp>
 #include <ddsrouter/writer/IWriter.hpp>
```
[the remark this answers is not in the document]
Comment (jparisu) [reply include/ddsrouter/writer/implementations/auxiliar/DummyWriter.hpp:29]: :+1:  (rebase fail)

File: src/cpp/participant/ParticipantFactory.cpp [hunk 6]
```diff
@@ -24,6 +24,7 @@
 #include <ddsrouter/participant/implementations/auxiliar/VoidParticipant.hpp>
 #include <ddsrouter/participant/ParticipantFactory.hpp>
 #include <ddsrouter/types/utils.hpp>
+#include <ddsrouter/types/Log.hpp>
```
[the remark this answers is not in the document]
Comment (jparisu) [reply src/cpp/participant/ParticipantFactory.cpp:27]: :+1:

File: include/ddsrouter/types/participant/ParticipantType.hpp [hunk 7]
```diff
@@ -26,22 +26,44 @@
...
+    /**
+     * @brief Whether this ParticipantType is valid
+     *
+     * @return true if valid. False otherwise
+     */
+    bool is_valid() const noexcept;
+
+    std::string to_string() const noexcept;
+
+    ParticipantTypeValue operator ()() const noexcept;
```
[the remark this answers is not in the document]
Comment (jparisu) [reply include/ddsrouter/types/participant/ParticipantType.hpp:65]: :+1:

File: test/blackbox/ddsrouter_core/TrivialTest.cpp [hunk 8]
```diff
@@ -22,6 +22,7 @@
 #include <ddsrouter/participant/implementations/auxiliar/DummyParticipant.hpp>
 #include <ddsrouter/types/RawConfiguration.hpp>
 #include <ddsrouter/types/utils.hpp>
+#include <ddsrouter/types/Log.hpp>
```
[the remark this answers is not in the document]
Comment (jparisu) [reply test/blackbox/ddsrouter_core/TrivialTest.cpp:25]: :+1:

File: src/cpp/types/participant/ParticipantId.cpp [hunk 9]
```diff
@@ -20,58 +20,63 @@
 #include <set>
 #include <sstream>
 
-#include <ddsrouter/types/ParticipantId.hpp>
+#include <ddsrouter/types/participant/ParticipantId.hpp>
```
[the remark this answers is not in the document]
Comment (jparisu) [reply src/cpp/types/participant/ParticipantId.cpp:23]: :+1:

File: test/unittest/configuration/ConfigurationTest.cpp (hunk 1 above)
Comment (jparisu) [reply test/unittest/configuration/ConfigurationTest.cpp:219]: :raccoon:

**Review by juanlofer-eprosima — APPROVED** [review]
LGTM

**Pull request merged**